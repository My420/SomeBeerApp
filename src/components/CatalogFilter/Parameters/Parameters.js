import React from 'react';
import PropTypes from 'prop-types';
import { TransitionGroup, CSSTransition } from 'react-transition-group';
import styles from './Parameters.module.scss';
import animation from './animation.module.css';
import withToggle from '../../HOC/ToggleWrapper/ToggleWrapper';
import InputFields from './InputFields/InputFields';
import Button from '../../Button/Button';
import {
  ABV_LESS_PROP,
  ABV_MORE_PROP,
  IBU_LESS_PROP,
  IBU_MORE_PROP,
  EBC_LESS_PROP,
  EBC_MORE_PROP,
  PAGE,
  BEER_NAME_PROP
} from '../../../utils/constants';

export const Parameters = ({
  options,
  onPropertyChange,
  isOpen,
  toggleState
}) => {
  const buttonText = isOpen ? 'hide' : 'options';
  return (
    <React.Fragment>
      <Button className={styles.button} onClick={toggleState}>
        <span className={styles.buttonText}>{buttonText}</span>
      </Button>
      <TransitionGroup component={null}>
        {isOpen ? (
          <CSSTransition classNames={{ ...animation }} timeout={300}>
            <div>
              <InputFields
                options={options}
                onPropertyChange={onPropertyChange}
              />
            </div>
          </CSSTransition>
        ) : null}
      </TransitionGroup>
    </React.Fragment>
  );
};

Parameters.propTypes = {
  options: PropTypes.shape({
    [PAGE]: PropTypes.string.isRequired,
    [BEER_NAME_PROP]: PropTypes.string.isRequired,
    [ABV_LESS_PROP]: PropTypes.string.isRequired,
    [ABV_MORE_PROP]: PropTypes.string.isRequired,
    [IBU_LESS_PROP]: PropTypes.string.isRequired,
    [IBU_MORE_PROP]: PropTypes.string.isRequired,
    [EBC_LESS_PROP]: PropTypes.string.isRequired,
    [EBC_MORE_PROP]: PropTypes.string.isRequired
  }).isRequired,
  onPropertyChange: PropTypes.func.isRequired,
  toggleState: PropTypes.func.isRequired, // from HOC
  isOpen: PropTypes.bool.isRequired // from HOC
};

export default withToggle(Parameters);
