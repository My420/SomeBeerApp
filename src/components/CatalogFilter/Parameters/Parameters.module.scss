@import '../../../sass/variables.scss';

.button {
  display: inline;
  padding: 0;

  background-color: transparent;
  border: none;

  text-decoration: underline;

  color: $second-text-color;
  @include text-button;
}

@media (max-width: 400px) {
  .button {
    font-size: 16px;
  }
}
