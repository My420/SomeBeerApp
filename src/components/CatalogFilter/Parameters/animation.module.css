.appear {
}
.appearActive {
}
.appearDone {
}
.enter {
  height: 0px;
  overflow: hidden;
}
.enterActive {
  height: 202px;
  transition: all 300ms;
}
.enterDone {
}
.exit {
  height: 202px;
  overflow: hidden;
}
.exitActive {
  height: 0px;
  transition: all 300ms;
}
.exitDone {
}

@media (max-width: 400px) {
  .enter {
    height: 0px;
    overflow: hidden;
  }
  .enterActive {
    height: 176px;
    transition: all 300ms;
  }

  .exit {
    height: 176px;
    overflow: hidden;
  }
  .exitActive {
    height: 0px;
    transition: all 300ms;
  }
}
