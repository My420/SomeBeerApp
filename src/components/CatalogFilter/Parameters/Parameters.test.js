import React from 'react';
import { shallow } from 'enzyme';
import { Parameters } from './Parameters';

describe('test <Parameters /> component', () => {
  describe('test <Parameters /> component with close InputFields', () => {
    const propFn = jest.fn();
    const toggleFn = jest.fn();
    const options = {};
    const wrapper = shallow(
      <Parameters
        options={options}
        onPropertyChange={propFn}
        isOpen={false}
        toggleState={toggleFn}
      />
    );

    beforeEach(() => {
      propFn.mockReset();
      toggleFn.mockReset();
    });

    test('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should call toggleState func on button click', () => {
      wrapper.find('Button').simulate('click');
      expect(toggleFn.mock.calls.length).toBe(1);
    });
  });

  describe('test <Parameters /> component with open InputFields', () => {
    const propFn = jest.fn();
    const toggleFn = jest.fn();
    const options = {};
    const wrapper = shallow(
      <Parameters
        options={options}
        onPropertyChange={propFn}
        isOpen
        toggleState={toggleFn}
      />
    );

    beforeEach(() => {
      propFn.mockReset();
      toggleFn.mockReset();
    });

    test('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('should call toggleState func on button click', () => {
      wrapper.find('Button').simulate('click');
      expect(toggleFn.mock.calls.length).toBe(1);
    });
  });
});
