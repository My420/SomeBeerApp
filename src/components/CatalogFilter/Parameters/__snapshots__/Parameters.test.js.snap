// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <Parameters /> component test <Parameters /> component with close InputFields should match snapshot 1`] = `
<Fragment>
  <Button
    activeStyle=""
    className="button"
    disabled={false}
    onClick={[MockFunction]}
  >
    <span
      className="buttonText"
    >
      options
    </span>
  </Button>
  <TransitionGroup
    childFactory={[Function]}
    component={null}
  />
</Fragment>
`;

exports[`test <Parameters /> component test <Parameters /> component with open InputFields should match snapshot 1`] = `
<Fragment>
  <Button
    activeStyle=""
    className="button"
    disabled={false}
    onClick={[MockFunction]}
  >
    <span
      className="buttonText"
    >
      hide
    </span>
  </Button>
  <TransitionGroup
    childFactory={[Function]}
    component={null}
  >
    <CSSTransition
      classNames={Object {}}
      timeout={300}
    >
      <div>
        <InputFields
          onPropertyChange={[MockFunction]}
          options={Object {}}
        />
      </div>
    </CSSTransition>
  </TransitionGroup>
</Fragment>
`;
